website/docs/language/values/variables.html.md

```diff
@@ -303,6 +304,23 @@ random_pet.animal: Creating...
 random_pet.animal: Creation complete after 0s [id=jae-known-mongoose]
 ```
 
+### Disallowing null Module Input Values
+
+[inpage-nullable]: #disallowing-null-input-values
+
+-> This feature was introduced in Terraform v1.1.0.
+
+The default value for `nullable` is `true`. When `nullable` is `true`, `null`
+is a valid value for the variable, and the module configuration must always
+account for the possibility of the variable value being `null`. Passing a
+`null` value as a module input argument will override any `default` value.
+
+Setting `nullable` to `false` ensures that the variable value will never be
+`null` within the module. If `nullable` is `false` and the variable has a
+`default` value, then the default will be used when a `null` value is given as
+a module input argument.
+
+
```

Comment:
Perhaps we should note here that the restriction applies only to the top-level value, and not to any nested values. Something like this?

> `nullable` only controls where the direct value of the variable may be `null`. For variables of collection or structural types, such as list or object types, the caller may still use `null` in nested elements or attributes.